eq-author-api/schema/tests/commentsConfirmationPage.test.js

```diff
@@ -0,0 +1,242 @@
+
+const { createQuestionnaire } = require("../../utils/datastore");
+
+const {
+  deleteQuestionnaire,
+} = require("../../tests/utils/contextBuilder/questionnaire");
+
+const {
+  queryComments,
+  createComment,
+  deleteComment,
+  updateComment,
+  createReply,
+  updateReply,
+  deleteReply,
+} = require("../../tests/utils/contextBuilder/comments");
+
+describe("comments", () => {
+  let ctx, questionnaire, confirmationId;
+
+  afterEach(async () => {
+    if (!questionnaire) {
+      return;
+    }
+    await deleteQuestionnaire(ctx, questionnaire.id);
+    questionnaire = null;
+  });
+
+  beforeEach(async () => {
+    ctx = await buildContext({
+      metadata: [{}],
+      sections: [
+        {
+          pages: [
+            {
+              confirmation: {},
+            },
+          ],
+        },
+      ],
+    });
+
+    questionnaire = ctx.questionnaire;
+    confirmationId = questionnaire.sections[0].pages[0].confirmation.id;
+  });
+
+  it("An empty comment array is created on new questionnare", async () => {
+    const comment = await queryComments(ctx, {
+      confirmationId: confirmationId,
+    });
+
+    expect(comment.confirmationPage).toMatchObject({
+      id: expect.any(String),
+      comments: expect.any(Array),
+    });
+  });
+
+  it("should add a comment on confirmationPage and then query that comment", async () => {
+    await createComment(ctx, {
+      confirmationId: confirmationId,
+      commentText: "a new comment is created",
+    });
+
+    const queryNewComments = await queryComments(ctx, {
+      confirmationId: confirmationId,
+    });
+
+    expect(queryNewComments.confirmationPage).toMatchObject({
+      id: confirmationId,
+      comments: [
+        {
+          commentText: "a new comment is created",
+        },
+      ],
+    });
+  });
+
+  it("should add multiple comments on confirmationPage and then query those comments", async () => {
+    await createComment(ctx, {
+      confirmationId: confirmationId,
+      commentText: "a new comment is created",
+    });
+
+    await createComment(ctx, {
+      confirmationId: confirmationId,
+      commentText: "a 2nd comment is created",
+    });
+
+    const queryNewComments = await queryComments(ctx, {
+      confirmationId: confirmationId,
+    });
+
+    expect(queryNewComments.confirmationPage).toMatchObject({
+      id: confirmationId,
+      comments: [
+        {
+          commentText: "a new comment is created",
+        },
+        {
+          commentText: "a 2nd comment is created",
+        },
+      ],
+    });
+  });
+
+  it("should edit a comment on confirmationPage and then query that comment", async () => {
+    const newComment = await createComment(ctx, {
+      confirmationId: confirmationId,
+      commentText: "a new comment is created",
+    });
+    const commentId = newComment.id;
+
+    const queryEditedComment = await updateComment(ctx, {
+      confirmationId: confirmationId,
+      commentId: commentId,
```

Comment:
You could reduce this object by doing the following

```suggestion
      commentId,
```